### PR TITLE
week 3 assessment update

### PR DESCRIPTION
**Pull request by ResetReesi**
# Submitting Week 3 Assessment

### ✅ Completed Files
Mark the following as confirmation that the files in this repository are complete.
- [ ✅] Coding Challenges
- [ ✅] Interview Questions
- [✅ ] React Debugging

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -11,46 +11,176 @@
...
+// ONLY WAS ABLE TO GET ON TEST TO PASS. SPENT THE WHOL WEEKEND ON JUST THIS
+
+
+let fibinochi= () => {}
+    const newArray = [0, 1];
+    for(var i = 2; i < newArray; i++) {
+       const a = (i - 1);
+       const b = (i - 1);
+       newArray.push(a + b);
+    }
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:112]: This is a great attempt and honestly it has all the ingredients for the recipe here! But putting the ingredients together correctly is also half the journey. 

Go and lookup function syntax in javascript unit 1 and check out what's going on with the curly brackets or "code block"

File: code-challenges.test.js [hunk 2]
```diff
@@ -11,46 +11,176 @@
...
+// TRY One  🤷🏾‍♀️
+const onlyNumbers = fullArr1.filter( 
+    element => typeof element === "number"
+); 
+
+const odds = onlyNumbers.filter(number => {
+    return number % 2 !== 0;
+});
+
+const sorted= odds.sort
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js]: A proper function should only be declared once and reused for each variable. Again check out whats going on with functions in javascript unit 1!

File: code-challenges.test.js [hunk 3]
```diff
@@ -11,46 +11,176 @@
...
+      let { sum, res } = acc;
+      sum += val;
+      res.push(sum);
+      return { sum, res };
+   }, {
+      sum: 0,
+      res: []
+   });
+   return accSum.res;
+};
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:221]: This is amazing code! I would love to see the pseudo code that explains this!

File: code-challenges.test.js [hunk 4]
```diff
@@ -11,46 +11,176 @@
...
+// // Expected output: [0, 7, -1, 11]
 
 const numbersToAdd3 = []
-// Expected output: []
+// // Expected output: []
+
+//PSUEDO CODE
+// create a functionn that takes in a an array
+// Input: numbersToAdd1,2,3, which are the original arrays given to us
+// Output: pushing numbers to a  array with the accumulation sum of the original array
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:181]: Okay but this pseudo code doesn't express **"how"** to solve this problem-- just what the problem is.

File: interview-questions.md [hunk 5]
```diff
@@ -10,46 +10,100 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+
+
+
+
+
 Researched answer:
 
+HTML is a Hypertext Markup Language, the standard language for documents designed to display and view on the web in a web browser. JSX (JavaScript + XML) is an extension of JavaScript that permits you to write down HTML directly within JavaScript, which features a few benefits of creating your code more readable and exercising the complete power of JavaScript within HTML. JSX is in some ways almost like HTML, but it does accompany certain differences. Since JSX isn’t a legitimate JAvascript code, it must be compiled into JS with a tool like Babel or similar. JSX is basically a fundamental way of how you define React components.JSX stands for JavaScript XML. It is a syntax extension for JavaScript that lets you use HTML tags right inside your JavaScript file. Instead of creating, configuring, and appending your HTML tags through JavaScript objects, you can create the elements using an XML-like syntax that will generate the DOM elements for you behind the scenes. Below is an example of JSX
+
+const App = <h1>Welcome to GeeksforGeeks</h1>;
```
Comment (WalkerTexas-coder) [inline interview-questions.md:23]: So please remember, even stitching together answers from multiple sites is still plagiarizing. 

[JSX (JavaScript + XML) is an extension of JavaScript that permits you to write down HTML directly within JavaScript, which features a few benefits of creating your code more readable and exercising the complete power of JavaScript within HTML.](https://www.geeksforgeeks.org/what-are-the-differences-between-jsx-and-html/#:~:text=Introduction%20to%20JSX%3A%20JSX%20(JavaScript,power%20of%20JavaScript%20within%20HTML. )

[What is JSX Exactly?
JSX stands for JavaScript XML. It is a syntax extension for JavaScript that lets you use HTML tags right inside your JavaScript file. ](https://www.webtips.dev/javascript-vs-jsx/)


[ h1 Welcome to GeeksforGeeks /h1 ;](https://www.geeksforgeeks.org/reactjs-functional-components/)

**Review by WalkerTexas-coder — CHANGES_REQUESTED** [review]
Okay Brook, there some work to do! I found where you copied your interview answers from. I am putting in a request for changes that you redo all of your interview answers and researched answers. They must be in your own words entirely and not copied at all, as per the requirements of this assessment. Once those changes are made alert me and I will re review this assessment. Also check out the feedback I gave you in the coding section.

**Comment by ResetReesi** [comment]
I will take everything I learned this week and  apply it to next week , so I can become a better developer.

**Pull request merged**